extensions/amp-form/0.1/amp-form.js

```diff
@@ -216,6 +221,29 @@ export class AmpForm {
   }
 
   /**
+   * Returns form data as an object.
+   * @return {!Object}
+   * @private
+   */
+  getFormAsObject_() {
+    const data = {};
+    const inputs = this.form_.querySelectorAll('input,select,textarea');
+    for (let i = 0; i < inputs.length; i++) {
+      const input = inputs[i];
+      if (!input.name || !input.value) {
```

Comment:
> (Maybe just piggyback on FormData to do the hard work here)
Unfortunately support for `FormData` is very limited when it comes to iterating over its fields. webkit for example only has the `.append` and the initialization of FormData from a `form` object. So there doesn't seem to be a way to do this for now.

> disabled inputs should also be excluded.
Good point. Will update the code for this.